test/k8sT/KafkaPolicies.go

```diff
@@ -0,0 +1,257 @@
+		Expect(err).Should(BeNil())
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			ciliumPod, _ := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			kubectl.CiliumReport(helpers.KubeSystemNamespace, ciliumPod, []string{
+				"cilium bpf tunnel list",
+				"cilium endpoint list",
+				"cilium service list"})
+		}
+
+		kubectl.Delete(demoPath)
+
+	})
+
+	waitUntilEndpointUpdates := func(pod string, eps map[string]int64, min int) error {
+		body := func() bool {
+			updated := 0
+			newEps := kubectl.CiliumEndpointPolicyVersion(pod)
+			for k, v := range newEps {
+				if eps[k] < v {
+					logger.Infof("Endpoint %s had version %d now %d updated : %d", k, eps[k], v, updated)
+					updated++
+				}
+			}
+			return updated >= min
+		}
+		err := helpers.WithTimeout(body, "No new version applied", &helpers.TimeoutConfig{Timeout: 100})
+		return err
+	}
+
+	getAppPods := func() map[string]string {
+		appPods := make(map[string]string)
+		apps := []string{helpers.KafkaApp, helpers.Zookeeper, helpers.Backup, helpers.EmpireHq, helpers.Outpost}
+		for _, v := range apps {
+			fmt.Printf("getting pod name of app: %s\n", v)
+			res, err := kubectl.GetPodNames(helpers.DefaultNamespace, fmt.Sprintf("app=%s", v))
+			Expect(err).Should(BeNil())
+			Expect(res).Should(Not(BeNil()))
+			fmt.Printf("res output: %q\n", res)
+			appPods[v] = res[0]
+			logger.Infof("KafkaPolicyRulesTest: pod=%q assigned to %q", res[0], v)
+		}
+		return appPods
+	}
+
+	It("KafkaPolicies", func() {
+		clusterIP, err := kubectl.Get(helpers.DefaultNamespace, "svc").Filter(
+			"{.items[?(@.metadata.name == \"kafka-service\")].spec.clusterIP}")
+		logger.Infof("PolicyRulesTest: cluster service ip '%s'", clusterIP)
+		Expect(err).Should(BeNil())
+
+		ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil())
+
+		status := kubectl.CiliumExec(ciliumPod, fmt.Sprintf("cilium config %s=%s", helpers.PolicyEnforcement, helpers.PolicyEnforcementDefault))
+		status.ExpectSuccess()
+		helpers.Sleep(5)
+
+		kubectl.CiliumEndpointWait(ciliumPod)
+
+		epsStatus := helpers.WithTimeout(func() bool {
+			endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
+			if err != nil {
+				return false
+			}
+			return endpoints.AreReady()
+		}, "Could not get endpoints", &helpers.TimeoutConfig{Timeout: 100})
+		Expect(epsStatus).Should(BeNil())
+
+		endpoints, err := kubectl.CiliumEndpointsListByLabel(ciliumPod, podFilter)
+		Expect(err).Should(BeNil())
+
+		Expect(endpoints.AreReady()).Should(BeTrue())
+
+		appPods := getAppPods()
+		By("Testing basic kafka setup")
+
+		_, err = kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[helpers.EmpireHq], fmt.Sprintf("-c \"echo “Happy 40th Birthday to General Tagge” | ./kafka-produce.sh --topic empire-announce\""))
+		Expect(err).Should(BeNil())
```

Comment:
How come the order of the produce / consume is different than in the GSG? In the GSG, the clients start to consume before the producer produces `empire-announce`.

Reply:
We need to produce first, since consumer script *wait* for something to be there. This test will hang if we consume first, since it will never return. This order is fine for this test

Reply:
OK, thanks.

Reply:
sure

Reply:
done. added a comment to be more clear.